ee/tabby-webserver/src/schema/gitlab_repository_provider.rs

```diff
@@ -0,0 +1,136 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use juniper::{GraphQLInputObject, GraphQLObject, ID};
+use lazy_static::lazy_static;
+use regex::Regex;
+use validator::Validate;
+
+use super::Context;
+use crate::{juniper::relay::NodeType, schema::Result};
+
+lazy_static! {
+    static ref GITLAB_REPOSITORY_PROVIDER_NAME_REGEX: Regex = Regex::new("^[\\w-]+$").unwrap();
+}
+
+#[derive(GraphQLInputObject, Validate)]
+pub struct CreateGitlabRepositoryProviderInput {
+    #[validate(regex(code = "displayName", path = "GITLAB_REPOSITORY_PROVIDER_NAME_REGEX"))]
+    pub display_name: String,
+    #[validate(length(code = "access_token", min = 10))]
+    pub access_token: String,
+}
+
+#[derive(GraphQLInputObject, Validate)]
+pub struct UpdateGitlabRepositoryProviderInput {
+    pub id: ID,
+    #[validate(regex(code = "displayName", path = "GITLAB_REPOSITORY_PROVIDER_NAME_REGEX"))]
+    pub display_name: String,
+    #[validate(length(code = "access_token", min = 10))]
+    pub access_token: String,
+}
+
+#[derive(GraphQLObject, Debug, PartialEq)]
+#[graphql(context = Context)]
+pub struct GitlabRepositoryProvider {
+    pub id: ID,
+    pub display_name: String,
+
+    pub connected: bool,
+
```

Comment:
```suggestion
```

as discussed in db schema, add a new enum field `status` with three values:
1. Ready: access_token != empty && connected.
2. Pending: access_token != empty && !connected.
3. Error: access_token == empty.

Same logic should add to github as well, but it can be done in a follow up pr